codec/src/test/java/io/netty/handler/codec/compression/ZstdDecoderTest.java

```diff
@@ -30,6 +32,11 @@ public EmbeddedChannel createChannel() {
 
     @Override
     protected byte[] compress(byte[] data) throws Exception {
-        return Zstd.compress(data);
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        ZstdOutputStream zstdOs = new ZstdOutputStream(os);
```

Comment:
Let's use `ZstdOutputStreamNoFinalizer` instead, to avoid allocating a `Finalizer` object every time.